### DISCUSSION TITLE
Augmented faces

### DISCUSSION THREAD

**Discussion by abbasalid** [sceneview/sceneview#479] [Ideas]
Can we add augmented faces support? Edit: Can you please add augmented faces support :)

**Comment by ThomasGorisse** [comment]
We are open to any PR on that.
Feel free to create one and we will help you with pleasure on it.

**Comment by ThomasGorisse** [comment]
ARCore's Augmented Faces API can be used with SceneView v4.0.0-rc.1 by configuring the session:

```kotlin
ARSceneView(
    sessionConfiguration = { session, config ->
        config.augmentedFaceMode = Config.AugmentedFaceMode.MESH3D
    },
    onSessionUpdated = { session, frame ->
        val faces = session.getAllTrackables(AugmentedFace::class.java)
        // Process face meshes and pose data
    }
)
```

Note that Augmented Faces requires the front camera. You can access the face mesh, regions (forehead, nose tip, etc.), and apply overlays. SceneView gives you the session access — the face tracking logic uses standard ARCore APIs.
